### PR TITLE
Add bits needed for s2i integration

### PR DESCRIPTION
**Pull request by fridex**
## Related Issues and Dependencies

Related: https://github.com/thoth-station/thoth-application/pull/809
Fixes: https://github.com/thoth-station/solver/issues/417
Related: https://github.com/thoth-station/s2i-thoth/issues/128

Might be better to make solver an s2i application to solve https://github.com/thoth-station/solver/issues/417

With this change, there is no need to maintain dockerfiles and we will fully utilize the build pipeline.

## This introduces a breaking change

- [x] No

### CODE REVIEW

File: .thoth.yaml [hunk 1]
```diff
@@ -1,14 +1,80 @@
...
 runtime_environments:
-  - name: rhel:8
+  - name: solver-fedora-30-py36
     operating_system:
-      name: rhel
+      name: "fedora"
+      version: "30"
+    python_version: "3.6"
+    recommendation_type: latest
+    base_image: quay.io/thoth-station/s2i-thoth-f30-py36:v0.23.2
```
Comment (fridex) [inline .thoth.yaml:13]: Some of the s2i base images are not available, we should provide them to have the whole workflow in place.

**Comment by fridex** [comment]
Builds tested without overlays directory in the test cluster. Looks like the build process works and is utilizing s2i thoth tooling.

/assign @harshad16

File: .thoth.yaml (hunk 1 above)
Comment (harshad16) [reply .thoth.yaml:13]: ++
yes, we should create them

**Review by harshad16 — COMMENTED** [review]
/lgtm

this is excellent, we can merge this in and wait to make the next release once other part like base image and the build pipeline gets ready.

**Comment by fridex** [comment]
> /lgtm
> 
> this is excellent, we can merge this in and wait to make the next release once other part like base image and the build pipeline gets ready.

Awesome, thanks! Re missing base images - do we want to build them anyway? It looks like there are missing old Fedora releases which we do not need to support anyway. WDYT?

**Pull request merged**